lib/openshift/replication_controller.rb

```diff
@@ -25,21 +25,21 @@ def status_reachable?(from_status, to_status)
 
     # @param status [Symbol, Array<Symbol>] the expected statuses as a symbol
     # @return [Boolean] if pod status is what's expected
-    # def status?(user:, status:, quiet: false, cached: false)
-    #   statuses = {
-    #     waiting: "Waiting",
-    #     running: "Running",
-    #     succeeded: "Succeeded",
-    #     failed: "Failed",
-    #     complete: "Complete",
-    #   }
-    #   res = describe(user, quiet: quiet)
-    #   if res[:success]
-    #     pods_status = res[:parsed][:pods_status]
-    #     res[:success] = (pods_status[status].to_i != 0)
-    #   end
-    #   return res
-    # end
+    def status?(user:, status:, quiet: false, cached: false)
+      statuses = {
+        waiting: "Waiting",
+        running: "Running",
+        succeeded: "Succeeded",
+        failed: "Failed",
+        complete: "Complete",
+      }
+      res = describe(user, quiet: quiet)
+      if res[:success]
+        pods_status = res[:parsed][:pods_status]
+        res[:success] = (pods_status[status].to_i != 0)
+      end
```

Comment:
What is the idea here? if we see any pod cound with desired status that we consider status reached? I'm not sure such logic will always apply.

I think this method has been commented out in the past because it was not well understood. Could you describe your use case and possibly scenario to use with so that we can figure out what is actually needed here. Whether we need such custom method or not. And with what implementation.

Reply:
It seems we need the variable here, it will failed with `undefined method downcase for nil:NilClass (NoMethodError)
` if we commented out.
```
  Given I wait until the status of "hooks-2" replication_controller becomes :failed # features/step_definitions/resources.rb:126
      undefined method `downcase' for nil:NilClass (NoMethodError)
      /home/hasha/verification-tests/lib/openshift/resource.rb:251:in `phase'
      /home/hasha/verification-tests/lib/openshift/resource.rb:264:in `status?'
      /home/hasha/verification-tests/lib/openshift/resource.rb:360:in `block in wait_till_status'
      /home/hasha/verification-tests/lib/base_helper.rb:153:in `wait_for'
      /home/hasha/verification-tests/lib/openshift/resource.rb:359:in `wait_till_status'
      /home/hasha/verification-tests/features/step_definitions/resources.rb:129:in `/^I wait until the status of "(.+?)" (\w+) becomes :(.+)$/'
      features/tierN/web/admin-console/deployments.feature:169:in `Given I wait until the status of "hooks-2" replication_controller becomes :failed
```
One checkpoint of the case need to cancel dc rollout and waiting the latest rc becomes failed status, then check console page shows the expected states on page.

Reply:
ok, it is fine to have the method. But it needs to make sense. Presently one variable is not used at all. Then one pod reaching proper status results in setting the status of the whole thing.

Or maybe we need to define the step better. See my other comment.